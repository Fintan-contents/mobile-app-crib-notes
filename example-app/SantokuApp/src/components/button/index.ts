export * from './Button';
export * from './FilledButton';
export * from './IconButton';
export * from './OutlinedButton';
