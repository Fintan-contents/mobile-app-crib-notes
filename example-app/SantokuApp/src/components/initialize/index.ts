export {WithInitializeContext, useInitializeContext} from './InitializeContext';
