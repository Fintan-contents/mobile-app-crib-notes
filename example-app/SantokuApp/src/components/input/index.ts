export * from './PasswordTextInput';
export * from './TextInput';
