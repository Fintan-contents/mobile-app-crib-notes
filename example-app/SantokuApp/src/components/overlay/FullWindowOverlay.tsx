import React from 'react';
import {Platform, View} from 'react-native';

export const FullWindowOverlay = (props: {children: React.ReactNode}) => {
  if (Platform.OS === 'ios') {
    // iOS以外の場合にreact-native-screensのFullWindowOverlayを使用すると警告が出るので、iOSの場合のみ使用します。
    // requireした場合の型はanyとなってしまいESLintエラーが発生しますが無視します。
    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
    const RNSFullWindowOverlay = require('react-native-screens').FullWindowOverlay as typeof View;
    return <RNSFullWindowOverlay testID="FullWindowOverlay">{props.children}</RNSFullWindowOverlay>;
  }
  return <>{props.children}</>;
};
