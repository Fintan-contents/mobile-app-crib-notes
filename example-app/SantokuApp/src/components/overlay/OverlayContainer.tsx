import {useWorkletCallback} from 'framework/utilities/useWorkletCallback';
import React from 'react';
import {StyleSheet, View, ViewProps} from 'react-native';
import Reanimated, {BaseAnimationBuilder, FadeIn, FadeOut, Keyframe} from 'react-native-reanimated';

export type ModalContainerProps = Omit<Reanimated.AnimateProps<ViewProps>, 'entering' | 'exiting'> & {
  isVisible: boolean;
  /**
   * iOSの場合、アニメーションが終わった後に呼び出されます。
   * Androidの場合、アニメーションが始まった時に呼び出されます。
   */
  enteringCallback?: (finished: boolean) => unknown;
  exitingCallback?: (finished: boolean) => unknown;
  /**
   * enteringに指定したAnimationBuilderなどでwithCallbackを指定しても、本コンポーネント内で上書きしているため実行できません。
   * withCallbackで実行する関数は、enteringCallbackで指定してください。
   */
  entering?: BaseAnimationBuilder | Keyframe;
  /**
   * exitingに指定したAnimationBuilderなどでwithCallbackを指定しても、本コンポーネント内で上書きしているため実行できません。
   * withCallbackで実行する関数は、exitingCallbackで指定してください。
   */
  exiting?: BaseAnimationBuilder | Keyframe;
};

export const MODAL_CONTAINER_DEFAULT_FADE_IN_DURATION = 300;
export const MODAL_CONTAINER_DEFAULT_FADE_OUT_DURATION = 300;
export const MODAL_CONTAINER_DEFAULT_ENTERING = FadeIn.duration(MODAL_CONTAINER_DEFAULT_FADE_IN_DURATION);
export const MODAL_CONTAINER_DEFAULT_EXITING = FadeOut.duration(MODAL_CONTAINER_DEFAULT_FADE_OUT_DURATION);

export const OverlayContainer: React.FC<ModalContainerProps> = ({
  isVisible,
  entering = MODAL_CONTAINER_DEFAULT_ENTERING,
  exiting = MODAL_CONTAINER_DEFAULT_EXITING,
  enteringCallback,
  exitingCallback,
  style,
  children,
  ...animatedViewProps
}) => {
  const composedEnteringCallback = useWorkletCallback(enteringCallback);
  const composedExitingCallback = useWorkletCallback(exitingCallback);

  return (
    <>
      {isVisible && (
        <View style={styles.container} pointerEvents="box-none">
          <Reanimated.View
            style={style}
            entering={entering.withCallback(composedEnteringCallback)}
            exiting={exiting.withCallback(composedExitingCallback)}
            pointerEvents="box-none"
            {...animatedViewProps}>
            {children}
          </Reanimated.View>
        </View>
      )}
    </>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'flex-end',
  },
});
