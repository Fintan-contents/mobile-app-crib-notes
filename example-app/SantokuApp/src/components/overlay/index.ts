export * from './snackbar';
export * from './loading';
export * from './WithOverlay';
export * from './OverlayBackdrop';
export * from './OverlayContainer';
