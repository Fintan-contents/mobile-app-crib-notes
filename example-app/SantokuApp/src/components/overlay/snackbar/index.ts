export * from './Snackbar';
export * from './WithSnackbar';
