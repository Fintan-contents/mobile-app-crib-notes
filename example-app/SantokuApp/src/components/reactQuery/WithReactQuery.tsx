import NetInfo from '@react-native-community/netinfo';
import React, {useCallback, useEffect, useMemo} from 'react';
import {AppState, AppStateStatus, Platform} from 'react-native';
import {focusManager, onlineManager, QueryClient, QueryClientProvider} from 'react-query';

import {useDefaultQueryCache, useDefaultMutationCache} from './useDefaultCache';
import {useDefaultOptions} from './useDefaultOptions';

const WithReactQuery: React.FC = ({children}) => {
  const queryCache = useDefaultQueryCache();
  const mutationCache = useDefaultMutationCache();
  const defaultOptions = useDefaultOptions();

  const queryClient = useMemo(() => {
    return new QueryClient({
      queryCache,
      mutationCache,
      defaultOptions,
    });
  }, [queryCache, mutationCache, defaultOptions]);

  const onAppStateChange = useCallback((newAppState: AppStateStatus) => {
    if (Platform.OS !== 'web') {
      focusManager.setFocused(newAppState === 'active');
    }
  }, []);

  useEffect(() => {
    // アプリがバックグラウンドからアクティブに変化した際にrefetchできるようにする
    const subscription = AppState.addEventListener('change', onAppStateChange);
    return () => subscription.remove();
  }, [onAppStateChange]);

  useEffect(() => {
    // オフライン状態からオンライン状態に変化した際にrefetchできるようにする
    onlineManager.setEventListener(setOnline => {
      return NetInfo.addEventListener(state => {
        setOnline(state.isConnected ?? false);
      });
    });
  }, []);

  return <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>;
};

export {WithReactQuery};
