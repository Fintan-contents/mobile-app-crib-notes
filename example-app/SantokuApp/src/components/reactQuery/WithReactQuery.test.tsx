import {render} from '@testing-library/react-native';
import {WithSnackbar} from 'components/overlay';
import {WithAccountContext} from 'context/WithAccountContext';
import React from 'react';
import {Text} from 'react-native';

import {WithReactQuery} from './WithReactQuery';

const Wrapper: React.FC = ({children}) => {
  const accountData = {account: {accountId: '123456789', deviceTokens: []}};
  return (
    <WithSnackbar>
      <WithAccountContext accountData={accountData}>{children}</WithAccountContext>;
    </WithSnackbar>
  );
};

describe('WithReactQuery', () => {
  test('子要素を正常にrenderできること', () => {
    const children = <Text testID="test">test</Text>;
    const result = render(<WithReactQuery>{children}</WithReactQuery>, {wrapper: Wrapper});
    expect(result.getByTestId('test')).not.toBeNull();
    expect(result).toMatchSnapshot();
  });
});
