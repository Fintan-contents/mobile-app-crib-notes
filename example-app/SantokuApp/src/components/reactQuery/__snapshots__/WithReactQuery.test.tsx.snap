// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithReactQuery 子要素を正常にrenderできること 1`] = `
Array [
  <Text
    testID="test"
  >
    test
  </Text>,
  ";",
]
`;
